LPD-28178 [green ci] Delete ClientExtensionFDSFilter.testcase

chestofwonders opened the pull request:
Issue: https://liferay.atlassian.net/browse/LPD-28178

<h1>Delete ClientExtensionFDSFilter.testcase</h1>

These tests have been migrated, the remaining ones have been removed:

LPS-190220. Verify that when the user clicks on Cancel, the changes are not saved.
LPS-190220. Verify that when the user fills all fields and clicks on Publish, the changes are saved.
LPS-190220. Verify that the user can Delete the FDS Filter entry
LPS-190220. Verify that the user can edit the CX Filter entry
LPS-190220. Verify that the JavaScript URL field is mandatory
LPS-190220. Verify that the Name field cannot be empty

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**

chestofwonders commented:
@dsanz @markocikos @juanjofgliferay I was not able to test the name internationalization button, as it was impossible to open it from PW. I have found that it is not the same one we use in the DataSet, so I understand that we would have to ask another team for a refactoring. Am I right?

juanjofgliferay commented:
@chestofwonders AFAIK, the CX page does not use Clay components (https://github.com/liferay/liferay-portal/blob/master/modules/apps/client-extension/client-extension-web/src/main/resources/META-INF/resources/admin/edit_client_extension_entry.jsp#L64) as we do in the DataSet page. The behavior of the button is completely different from the one we are using in terms of accessibility (button lacks `aria-controls` and many others).
Also, the dropdown component with the languages is positioned in a different way.

juanjofgliferay commented:
@chestofwonders using the `clickAndExpectToBeVisible` could do the trick :wink: 

```
	await clickAndExpectToBeVisible({
		autoClick: true,
		target: page.locator('.input-localized-palette-container'),
		trigger:
			clientExtensionsPage.newCustomElementFormModal
				.localizedNameButton,
	});
```

![image](https://github.com/user-attachments/assets/e7b12a29-68bc-482c-a42e-c0558309eb39)

juanjofgliferay commented:
Overall, PR LGTM but I was wondering why you are creating those tests in the `customElement.spec.ts` file instead of using a generic `clientExtensionAdmin.spec.ts` one or a specific `fdsFilter.spec.ts` one.

chestofwonders commented:
> @chestofwonders using the `clickAndExpectToBeVisible` could do the trick 😉
> 
> ```
> 	await clickAndExpectToBeVisible({
> 		autoClick: true,
> 		target: page.locator('.input-localized-palette-container'),
> 		trigger:
> 			clientExtensionsPage.newCustomElementFormModal
> 				.localizedNameButton,
> 	});
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/132653342/379202429-e7b12a29-68bc-482c-a42e-c0558309eb39.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk2NzY1MDcsIm5iZiI6MTcyOTY3NjIwNywicGF0aCI6Ii8xMzI2NTMzNDIvMzc5MjAyNDI5LWU3YjEyYTI5LTY4YmMtNDgyYy1hNDJlLWMwNTU4MzA5ZWIzOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyM1QwOTM2NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMjg1YWNhNGM5MzFlM2EwMWEzYmRiNGJkMGVlNzZjNmFiZTNlMjk2NWE3YzdmOTFjYzhjNzVhNWIwYjc2NmQwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.cyoU2JrNon1mbx0a_zu618cd1N4nISLerOVDjzFiUL4)

Yes, it does. Love you!

chestofwonders commented:
> Overall, PR LGTM but I was wondering why you are creating those tests in the `customElement.spec.ts` file instead of using a generic `clientExtensionAdmin.spec.ts` one or a specific `fdsFilter.spec.ts` one.

I thought they were better packaged into customElement.spec.ts because the differences between CX types makes a generic admin test less usable than the individual types 😗

juanjofgliferay commented:
> > Overall, PR LGTM but I was wondering why you are creating those tests in the `customElement.spec.ts` file instead of using a generic `clientExtensionAdmin.spec.ts` one or a specific `fdsFilter.spec.ts` one.
> 
> I thought they were better packaged into customElement.spec.ts because the differences between CX types makes a generic admin test less usable than the individual types 😗

To solve this, we could use the same pattern used in the [FiltersPage](https://github.com/liferay/liferay-portal/blob/master/modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-admin/pages/data_set/tabs/FiltersPage.ts#L29) where each different form extends from a shared one. No need to create all in one shot.

dsanz commented:
Yep that makes sense. @chestofwonders pls check if you can do this now, in a separate subtask or in a different task we'll plan later. This may affect tests from other teams too as the `client-extension-web` pw folder contains many

chestofwonders commented:
ci:test:sf

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-40.liferay.com/job/test-portal-source-format/6530/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-40.liferay.com/userContent/jobs/test-portal-source-format/builds/6530/jenkins-report.html">here</a>.</b></p>

chestofwonders commented:
> Yep that makes sense. @chestofwonders pls check if you can do this now, in a separate subtask or in a different task we'll plan later. This may affect tests from other teams too as the `client-extension-web` pw folder contains many

I've rather leave that for another ticket so as not to delay this PR any further.
 
I have updated it with the test to check the internationalization and I have added some fixes for the validations, since it did not always check the required fields correctly. So, please @dsanz @juanjofgliferay @markocikos check again if you see any complains to forward it

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-40.liferay.com/job/test-portal-source-format/6530/">test-portal-source-format#6530</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-40.liferay.com/userContent/jobs/test-portal-source-format/builds/6530/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/4506">liferay-frontend#4506</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357/build/52336110">[master] ci:test:sf - chestofwonders &gt; liferay-frontend - PR#4506 - 2024-10-25[07:02:10]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>52336110<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-40.liferay.com/job/publish-testray-report/24983/">publish-testray-report#24983</a><br/></div></div>

juanjofgliferay commented on modules/test/playwright/tests/client-extension-web/pages/ClientExtensionsPage.ts:
```diff
@@ -31,6 +63,37 @@ export class ClientExtensionsPage {
+			}),
+			localizedNameButton: page
+				.locator('.input-localized')
+				.getByRole('button'), // page.getByRole('button', { name: 'Current translation is' }),
```
:broom:

chestofwonders replied:
removed!

juanjofgliferay commented on modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/derived-config.txt:
```diff
@@ -0,0 +1,586 @@
+com/liferay/frontend/taglib/clay/sample/web/internal/display/context/ClaySampleImageCard.lambda$getLabels$7(Lcom/liferay/frontend/taglib/clay/servlet/taglib/util/LabelItem;)V
```
@chestofwonders Curious abouth what is the purpose of this file?

chestofwonders replied:
Commited by mistake. It's generated by gradle

juanjofgliferay reviewed (approved):
LGTM

chestofwonders commented:
ci:forward

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

The pull request will automatically be forwarded to the user `brianchandotcom` If the following test suites pass:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>relevant</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest(master)/6309/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-35.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/6309/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<h1>Build completed.</h1><p>Jenkins URL: <a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest(master)/6309/">test-portal-acceptance-pullrequest(master)</a></p><p>The <code>relevant</code> test suite did not execute tests because all modified files were excluded by the <b>modified.files.global.excludes</b> property in <code>test.properties</code>.</p><p>Please use <code>ci:test:sf</code> and/or <code>ci:forward</code> for this pull request to be forwarded or eligible for forwarding.</p>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-5.liferay.com/job/test-portal-source-format/6530/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-5.liferay.com/userContent/jobs/test-portal-source-format/builds/6530/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest(master)/6309/">test-portal-acceptance-pullrequest(master)#6309</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-35.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/6309/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>relevant<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/4506">liferay-frontend#4506</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357/build/53627368">[master] ci:test:relevant - chestofwonders &gt; liferay-frontend - PR#4506 - 2024-10-28[03:06:35]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>53627368<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-35.liferay.com/job/publish-testray-report/13901/">publish-testray-report#13901</a><br/></div></div>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 7 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/407a4b2008251d4547096e56a868dbdde6c81ba1">407a4b2008251d4547096e56a868dbdde6c81ba1</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/chestofwonders/liferay-portal/tree/LPD-28178">LPD-28178</a><br/>Branch GIT ID: <a href="https://github.com/chestofwonders/liferay-portal/commit/c1439cd9608ee86aea821c1c5265a3808c7f2339">c1439cd9608ee86aea821c1c5265a3808c7f2339</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-5.liferay.com/job/test-portal-source-format/6530/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-5.liferay.com/userContent/jobs/test-portal-source-format/builds/6530/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
All required test suite(s) passed.
Forwarding pull request to `brianchandotcom`.
[Console](https://test-1-5.liferay.com/job/test-portal-source-format/6530//consoleFull)

liferay-continuous-integration commented:
Pull request has been successfully forwarded to  https://github.com/brianchandotcom/liferay-portal/pull/155625
[Console](https://test-1-5.liferay.com/job/test-portal-source-format/6530//consoleFull)

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-5.liferay.com/job/test-portal-source-format/6530/">test-portal-source-format#6530</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-5.liferay.com/userContent/jobs/test-portal-source-format/builds/6530/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/4506">liferay-frontend#4506</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357/build/53627420">[master] ci:test:sf - chestofwonders &gt; liferay-frontend - PR#4506 - 2024-10-28[03:06:34]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>53627420<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-5.liferay.com/job/publish-testray-report/3355/">publish-testray-report#3355</a><br/></div></div>
